component/DashBoard/DashBoard.jsx

```diff
@@ -85,249 +66,216 @@ function DashBoard({ user }) {
   }
   async function fetchMonthlyLeaderBoardData() {
     await httpService
-      .post(`/api/dashboard/monthly_leaderboard`, {
+      .post(`/api/dashboard/leaderboard`, {
         date: currentMonth,
         userId: user.id,
       })
       .then(({ data: response }) => {
         if (response.status === 200) {
-          setMonthlyLeaderBoardData(response.data);
-        }
-      })
-      .catch((err) => {});
-  }
-
-  async function fetchFeedbackData() {
-    await httpService
-      .post(`/api/feedback/all`, {
-        currentMonth: currentMonth,
-        userId: user.id,
-      })
-      .then(({ data: response }) => {
-        if (response.status === 200) {
-          setFeedbackList(response.data);
+          setMonthlyLeaderBoardData({
+            leaderBoardData: response.data,
+            leaderboardLoading: false,
+          });
         }
       })
       .catch((err) => {
-        setFeedbackList([]);
+        setMonthlyLeaderBoardData({
+          leaderBoardData: [],
+          leaderboardLoading: false,
+        });
       });
   }
 
-  async function fetchApplaudData() {
-    await httpService
-      .post(`/api/applaud/all`, {
-        date: currentMonth,
-      })
-      .then(({ data: response }) => {
-        if (response.status === 200) {
-          let data = response?.data?.sort(
-            (a, b) =>
-              b[Object.keys(b)]?.taken?.length -
-              a[Object.keys(a)]?.taken?.length
-          );
-          setAllApplaud(data);
-        }
-      })
-
-      .catch((err) => {});
-  }
-
   useEffect(() => {
     fetchDashboardData();
-    fetchFeedbackData();
-    fetchApplaudData();
     fetchMonthlyLeaderBoardData();
   }, []);
 
-  return (
-    <Row gutter={[24, 24]}>
-      <Col sm={24} md={24} lg={16} xxl={18}>
-        <div className="container mx-auto max-w-full space-y-6">
-          <div className="grid grid-cols-1 md:grid-cols-3 gap-4 2xl:gap-6">
-            <Link href={URLS.REVIEW_CREATED} passHref>
-              <>
-                <CountCard
-                  title={"Review Created"}
-                  count={dashBoardData.reviewCreatedCount}
-                  className="cursor-pointer"
-                  Icon={() => <SmallApplaudIcon />}
-                  tooltipText="Count of Reviews Created by you."
-                />
-              </>
-            </Link>
+  const sortGoalListByEndDate = useMemo(() => {
+    if (Number(goalList?.length) > 0) {
```

Comment:
explain this statement.

Reply:
if goalList is undefined than Number(goalList?.length) will be NaN which will be false